### PR TITLE
Feat: new Custom role resource

### PR DESCRIPTION
**Pull request by TomerHeber**
### Solution

Part of #498 (not yet complete - need to add examples and integration tests).
1. Added the resource.
2. Added unit tests.

### CODE REVIEW

File: env0/resource_user_organization_assignment.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		DeleteContext: resourceUserOrganizationAssignmentDelete,
+
+		Description: "note: if removed the user role will be reset to 'Organization User'",
+
+		Schema: map[string]*schema.Schema{
+			"user_id": {
+				Type:        schema.TypeString,
+				Description: "id of the user",
+				Required:    true,
+				ForceNew:    true,
```
Comment (TomerHeber) [inline env0/resource_user_organization_assignment.go:30]: If user is changed, remove from existing and add to new one.

File: env0/resource_user_organization_assignment.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	role, ok := d.GetOk("role")
+	if !ok {
+		role = d.Get("custom_role_id")
+	}
+
+	if err := apiClient.OrganizationUserUpdateRole(userId, role.(string)); err != nil {
+		return diag.Errorf("failed to update user role organization: %v", err)
+	}
+
+	d.SetId(userId)
```
Comment (TomerHeber) [inline env0/resource_user_organization_assignment.go:99]: I've set the id to be the userId.
Each user has one role in the organization.

File: env0/resource_user_organization_assignment.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+	return nil
+}
+
+func resourceUserOrganizationAssignmentDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	userId := d.Get("user_id").(string)
+
+	if err := apiClient.OrganizationUserUpdateRole(userId, "User"); err != nil {
```
Comment (TomerHeber) [inline env0/resource_user_organization_assignment.go:109]: There's no such thing as deleting a role. If this is removed, set him to "User".
Added it to the description.

File: env0/resource_user_organization_assignment_test.go [hunk 4]
```diff
@@ -0,0 +1,239 @@
+package env0
```
Comment (TomerHeber) [inline env0/resource_user_organization_assignment_test.go:1]: integration tests and examples to follow in a follow-up PR.

**Review by GiliFaroEnv0 — APPROVED** [review]
Nice!
as already said need to add tests

**Pull request merged**